File: frontend/src/domain/publicationStatus.ts
export type PublicationStatus = {
  publicationStatus?: PublicationState
  withError?: boolean
}

export enum PublicationState {
  PUBLISHED = "PUBLISHED",
  UNPUBLISHED = "UNPUBLISHED",
  PUBLISHING = "PUBLISHING",
  DUPLICATED = "DUPLICATED",
  LOCKED = "LOCKED",
  DELETING = "DELETING",
  EXTERNAL_HANDOVER_PENDING = "EXTERNAL_HANDOVER_PENDING",
}

export const Label: Record<PublicationState, string> = {
  [PublicationState.PUBLISHED]: "Veröffentlicht",
  [PublicationState.UNPUBLISHED]: "Unveröffentlicht",
  [PublicationState.PUBLISHING]: "In Veröffentlichung",
  [PublicationState.DUPLICATED]: "Dublette",
  [PublicationState.LOCKED]: "Gesperrt",
  [PublicationState.DELETING]: "Löschen",
  [PublicationState.EXTERNAL_HANDOVER_PENDING]: "Fremdanlage",
}

File: backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/PublicationStatus.java
package de.bund.digitalservice.ris.caselaw.domain;

public enum PublicationStatus {
  EXTERNAL_HANDOVER_PENDING("Fremdanlage"),
  UNPUBLISHED("Unveröffentlicht"),
  PUBLISHED("Veröffentlicht"),
  PUBLISHING("In Veröffentlichung"),
  DELETING("Löschen"),
  LOCKED("Gesperrt"),
  DUPLICATED("Dublette"),
  UNKNOWN("Unbekannt");

  public final String deTranslation;

  PublicationStatus(String deTranslation) {
    this.deTranslation = deTranslation;
  }
}
